File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <math.h>
#include <algorithm>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> day;
    int size = progresses.size();
    int k = 0;
    
    for(int i = 0; i<size;i++){
        float tmp = ceil((100-progresses[i])/(float)speeds[i]);
        day.push((int)tmp);
    }


    while(!day.empty()){
        int cnt = 1;
        int n = day.front(); 
        day.pop();
        while(!day.empty()&&day.front()<=n) {
            cnt++;
            day.pop();
        }

        answer.push_back(cnt);

    }

    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []
    queue = deque([[i, j] for i, j in zip(progresses,speeds)])    
    
    while(queue):
        k = 0
        count = 0
        while(queue[0][0] < 100):
            count = count + 1
            queue[0][0] = queue[0][0]+queue[0][1]
        
        queue.popleft()
        k = k + 1
        
        for i in range(len(queue)):
            queue[i][0] = queue[i][0]+queue[i][1]*count
            
        while(queue):
            if queue[0][0] >= 100:
                queue.popleft()
                k = k + 1
            else: 
                answer.append(k)
                break
                
    answer.append(k)
            
    return answer